libraries/AP_BattMonitor/AP_BattMonitor_Backend.cpp

```diff
@@ -91,3 +91,128 @@ void AP_BattMonitor_Backend::update_resistance_estimate()
     // update estimated voltage without sag
     _state.voltage_resting_estimate = _state.voltage + _state.current_amps * _state.resistance;
 }
+
+float AP_BattMonitor_Backend::voltage_resting_estimate() const
+{
+    // resting voltage should always be greater than or equal to the raw voltage
+    return MAX(_state.voltage, _state.voltage_resting_estimate);
+}
+
+AP_BattMonitor::BatteryFailsafe AP_BattMonitor_Backend::update_failsafes(void)
```

Comment:
Wondering about this type name... `AP_BattMonitor::BatteryFailsafeState?`

Reply:
Preexisting, not touching now.